res/res_pjsip_sdp_rtp.c

```diff
@@ -1539,6 +1543,44 @@ static void set_session_media_remotely_held(struct ast_sip_session_media *sessio
 	}
 }
 
+
+/*! \brief Function which checks for RED in a session and initalize it if needed */
+static void check_red_support(struct ast_sip_session *session, struct ast_sip_session_media *session_media,
+	struct ast_format *format_parsed)
+{
+	/*
+	 * To keep API compatibility, we are using an array here, even if the red codec
+	 * is using only an integer internally, similar as the other codec attributes.
+	 * The rtp red support from asterisk expect the primary payload also added to
+	 * the array, therefore we need one more element.
+	 */
+	int red_pt_array[AST_RED_MAX_GENERATION+1];
+	memset(red_pt_array, 0, sizeof(red_pt_array));
+
+	if (session == NULL || session_media == NULL || format_parsed == NULL) {
+		ast_log(LOG_ERROR, "Invalid parameter for initializing RED\n");
+		return;
+	}
+
+	if (ast_format_cap_has_type(session->endpoint->media.codecs, AST_MEDIA_TYPE_TEXT)) {
+		if(strncasecmp(ast_format_get_name(format_parsed), "RED", 3) == 0){
+			int red_payload = *(int*)ast_format_attribute_get(format_parsed, "red_payload");
+			int red_num_gen = *(int*)ast_format_attribute_get(format_parsed, "red_num_gen");
```

Comment:
If the format_attr module isn't loaded - what happens here?

Reply:
Same as above, lets add another check here